photosynthesis_metrics/psnr.py

```diff
@@ -0,0 +1,55 @@
+r""" This module implements Peak Signal-to-Noise Ration (PSNR) in PyTorch.
+"""
+import torch
+from typing import Optional, Union
+
+from photosynthesis_metrics.utils import _validate_input
+
+
+def psnr(prediction: torch.Tensor, target: torch.Tensor,
+         data_range: Union[int, float] = 1.0, reduction: Optional[str] = 'mean'):
+    r"""Compute Peak Signal-to-Noise Ration for a batch of images.
+    Supports both greyscale and color images with RGB channel order.
+
+    Args:
+        prediction: Batch of predicted images with shape (batch_size x channels x H x W)
+        target: Batch of target images with shape  (batch_size x channels x H x W)
+        data_range: Value range of input images (usually 1.0 or 255). Default: 1.0
+        reduction: Reduction over samples in batch: "mean"|"sum"|"none"
+        
+    Returns:
+        PSNR: Index of similarity betwen two images.
+    Note:
+        Implementaition is based on Wikepedia https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio
+        Colour images are first converted to YCbCr format and only luminance component is considered.
+    """
+    _validate_input((prediction, target), allow_5d=False)
+
+    # Constant for numerical stability
+    EPS = 1e-8
+
+    if data_range == 255:
+        prediction = prediction / 255.
+        target = target / 255.
+
+    # Convert RGB image to YCbCr and take luminance: Y = 0.299 R + 0.587 G + 0.114 B
```

Comment:
It is not a common way to compute PSNR. Usually, PSNR on RGB images means -10 log(mse) where mse is computed across all three channels.

Reply:
PSNR is historically defined only for greyscale images. There is no consensus how to do it for RGB, I've seen both approaches (compute on luminance and compute on each channel separately). 
I'll add a flag to compute it on all 3 channels

Reply:
If you open any repo with RGB image enhancement problem (SR, Denoising, etc..) or any challenge (like NTIRE, AIM, etc.), in 99.99% cases PSNR will be computed on 3 channels. So, this definitely should be the default behavior.

Reply:
Added flag `convert_to_greyscale` which is `False` by default.
Now results match other implementations